numba/unicode.py

```diff
@@ -745,6 +745,33 @@ def concat_impl(a, b):
         return concat_impl
 
 
+@overload(operator.mul)
+def unicode_repeat(a, b):
+    if isinstance(a, types.UnicodeType) and isinstance(b, types.Integer):
+        def repeat_impl(a, b):
+            new_length = a._length * b
```

Comment:
At present, if the integer supplied is `<0` this will segfault. I think the integer needs to be wrapped like `repeats = max(int_value, 0)` and then `repeats` used throughout. Further, short-cutting for empty string, and integer values (0, 1) is probably worth it?

Reply:
oops, I didn't consider the negative case. I fixed it now, and added the shortcuts.